### DISCUSSION TITLE
Songs without an album metadata tag shouldn't be sorted into albums called "[Unknown Album]"

### DISCUSSION THREAD

**Discussion by abstractionmage** [navidrome/navidrome#3266] [Ideas]
Whenever Navidrome detects a song without an album tag in its metadata, it treats this song as being in an album called "[Unknown Album]" by whoever artist had made that particular song (see screenshot below). I don't know the original purpose of that behavior, but it causes my music collection, which is filled with songs made by hobbyist music producers that simply aren't part of an album, to become flooded with albums called "[Unknown Album]". In my collection 100 albums out of 420 are called "[Unknown Album]" at the moment, which makes browsing through actual albums less enjoyable and the only "solution" I found is to add fake album metadata tags to each and every song that doesn't have an album tag to begin with - but even then I end up with way too many albums than I actually have.

That's why I would prefer it if Navidrome wouldn't sort songs without album metadata into albums at all or if there was an option for me to turn that off.

![navidrome unknown album](https://github.com/user-attachments/assets/02333f73-5542-436c-97fa-20ecd7a6ade4)

![2024-09-15-204747_hyprshot](https://github.com/user-attachments/assets/84577587-5dc6-4708-8004-7738b0104eff)

**Comment by deluan** [comment]
I agree this is not the best solution, but there's no one-size-fits-all solution in this case. What do you propose? I mean, you asked for an option to turn it off, but then how would you find these songs? Only in the songs view?

**Reply by abstractionmage** [reply]
Yeah, the ideal solution _for me_ would be to only see these songs only in the "Songs" view. I mean I have plenty of albums and EPs in my collection, too, which I sometimes like to browse through in the various albums views ("Albums - Random" being my favorite one), but I don't really want to see any "Unknown Album"s with random songs when I do that. Only real albums that I purchased on Bandcamp with a matching title and cover art.

If I want to listen to songs that aren't in an album, I'd normally go over to the "Songs" view. Or I'd go to the "Artist" view and click on an artist whose work I'd like to listen to, in which case I would expect those songs to show up there as well.

Maybe that's just me, though. I haven't met a lot of other music collectors so far, so I don't know if others would share my perspective or not.

**Reply by dweymouth** [reply]
One thing to note about this is that if you use 3rd party clients eg mobile apps, there is no way for non-album songs to be reported via the API, so they would just be completely unavailable in 3rd party clients. Possibly it could be a Navidrome webUI option only to hide the "Unknown Albums" from the album list, rather than at the backend/API level

**Reply by mist2t** [reply]
Fixing this will bring music libraries to modern times, breaking an old-school habbit where every track must be tied to an album.

We can pretty much see on streaming services how this old design played in modern times: every new singles are released as a single track album ...

I would LOVE a fix that looked like this:

- treat any tracks without an album as "singles"
- dont automatically assign singles to a "unknown album"
- in albums browse pages show only "real" albums (those with the album tag specified)
- in artist profile show only "real" albums (those with the album tag specified)
- implement a "song" list in artist profile that will show all songs (including singles)
- optional idea: implement a "singles" list in artist profile that will show only singles (those tracks without an album tag)
- of course, all these "singles" will be listed in the general navidrome "Songs" browsing page alongside all the tracks in the whole library (the current functionality)

What do we win by this:
- ability to have only complete albums listed as actual albums
- user can choose to curate it's library and keep only few tracks (only the "hits")from albums and have those tracks as "singles"
- the library can function in modern times where artists mostly release lone singles.
- one can choose to only have a "singles" oriented library without any albums at all ... having all those tracks shown individualy in artist profile page.

Anyway, long story short ... i would LOVE a solution for this !

**Reply by deluan** [reply]
Yeah, I like this idea and I've been struggling to find a good solution for this. But without Subsonic API not explicitly supporting this, these singles would still need to be exposed to Subsonic clients as albums.

Also, "single" is not really what this is, as there are real released singles (7" and 12", even CD Singles) that may contain more than 1 song. This could cause confusion.

Anyway, even if this is implemented in the server, I don't see the required UI changes being done in the current UI, but I will definitelly consider this for the UI rewrite (coming after #2709)

**Reply by michaelkrieger** [reply]
> One thing to note about this is that if you use 3rd party clients eg mobile apps

The good news is @deluan is a part of the [open subsonic api working group](https://github.com/opensubsonic/open-subsonic-api) and these kinds of things can eventually make their way to client apps

**Comment by michaelkrieger** [comment]
See also [my issue ](https://github.com/navidrome/navidrome/discussions/2853) which has lots of discussion on the same topic. @deluan suggested NFO files as the appropriate solution to override.  Can’t come soon enough as it’s a big thing missing for me to import my library.

the main use cases I see is if you have a folder of misc files of different artists, albums, track info, dates. Like all your singles from 1999. You don’t want to edit out important information on the file, but you also don’t want the album to be in searches (though great if it shows when playing).  If you download a whole Apple Music or Spotify playlist to a folder, you’ll have a folder like this of misc singleton tracks that have nothing to do with each other.  Another example is the Billboard Top 40 for a week. Again a bunch of dissimilar songs in a folder and the album info isn’t really wanted to separate them out

**Reply by mist2t** [reply]
Yes, this is the "modern" type of music library 🙂, "spotify" style, playlist oriented ... where the whole library is basically a big compilation of singletons of various genres and artists.

Most of the time, for this type of use case, "albums" are just annoying since most of them are incomplete ... and browsing "by album" is totally irrelevant.

For this case there is also the oportunity to completely hide "albums" functionality globally in settings and let the whole library be "singletons" only.

Maybe after the new scanner is released and @deluan is starting to work on the new UI he will think about a solution to accomodate all use cases.

As i can see it, there are 3 cases for libraries:

- Albums oriented library where each track belongs to an album (current state)

- Singletons oriented library with no albums at all

- A mix of the two above (this is my use case)
